src/NumericalAlgorithms/Spectral/Projection.hpp

```diff
@@ -14,20 +14,60 @@ class Mesh;
 
 namespace Spectral {
 
+/// The portion of an element covered by a child element.
+enum class ChildSize { Full, UpperHalf, LowerHalf };
+
 /// The portion of an element covered by a mortar.
-enum class MortarSize { Full, UpperHalf, LowerHalf };
+using MortarSize = ChildSize;
 
-std::ostream& operator<<(std::ostream& os, MortarSize mortar_size) noexcept;
+std::ostream& operator<<(std::ostream& os, ChildSize mortar_size) noexcept;
 
 /*!
- * \brief The projection matrix from a 1D mortar to an element.
+ * \brief The projection matrix from a child element to its parent.
  *
- * \details
  * The projection matrices returned by this function (and by
- * projection_matrix_element_to_mortar()) define orthogonal projection
- * operators between the spaces of functions on the boundary of the
- * element and the mortar.  These projections are usually the correct
- * way to transfer data onto and off of the mortars.
+ * projection_matrix_parent_to_child()) define orthogonal projection operators
+ * between the spaces of functions on a parent element and its children. These
+ * projections are usually the correct way to transfer data between elements in
+ * a mesh-refinement hierarchy, as well as between an element face and its
+ * adjacent mortars.
+ *
+ * These functions assume that the `child_mesh` is at least as fine as the
+ * `parent_mesh`, i.e. functions on the `parent_mesh` can be represented exactly
+ * on the `child_mesh`. In practice this means that functions can be projected
+ * to a mortar (the `child_mesh`) from both adjacent element faces (the
+ * `parent_mesh`) without losing accuracy. Similarly, functions in a
+ * mesh-refinement hierarchy don't lose accuracy when an element is split
+ * (h-refined). For this reason, the `projection_matrix_child_to_parent` is
+ * sometimes referred to as a "restriction operator" and the
+ * `projection_matrix_parent_to_child` as a "prolongation operator".
+ *
+ * The projections implemented here are \f$L^2\f$ projections, or "Galerkin"
+ * projections. They are defined such that the projection from a parent to one
+ * of its children ("prolongation") is an (exact) interpolation, and the
+ * projection from a child to its parent ("restriction") is its inverse.
```

Comment:
I think "adjoint" is the correct word here

Reply:
It's been a long time since I did much linear algebra, but it doesn't seem to match any of the many meanings of "adjoint" listed on Wikipedia.  I think it is the "Moore–Penrose inverse" (or often "pseudoinverse").

Reply:
I'm not sure exactly how the pseudoinverse (that is what it is called in the subcell literature) is calculated, in DG-subcell it is a constrained linear-least squares problem with the constraint being conservation.

Reply:
Here's Wikipedia on "adjoint": https://en.wikipedia.org/wiki/Adjoint

> In mathematics, the term adjoint applies in several situations. Several of these share a similar formalism: if A is adjoint to B, then there is typically some formula of the type
>
> (Ax, y) = (x, By).

That seems pretty fitting to me :) I'm not entirely sure if this is the Moore-Penrose inverse since that's uniquely defined, though it might very well be.

Reply:
You can claim it's *an* adjoint, then :).  More seriously, it looks like "the adjoint" most commonly means the transpose, so you shouldn't say that.  I, too, am unsure about whether this is the pseudoinverse.  I also don't see anything useful for giving a named relation between the matrices in Saul's document.

Reply:
R is definitely a left inverse of P in the no-h-refinement case, because in spectral space they are both identity matrices with extra zeros tacked on.  I think in the h-refined case it is something like `R_L P_L + R_R P_R = id`, so not inverse.

(I continue to be concerned about claims that we are losing conservation by doing subcell.  If we don't care about conservation there is no reason to use an expensive time stepper.)

Reply:
I'm not sure where I said we lose conservation on subcells in the above, but this issue is present already in DG+AMR as far as I can tell because you need the Jacobian in the integrand. Currently with subcell I project `u` instead of `uJ`, so it does lose conservation. However, this loss of conservation occurs where the solution is smooth, and so strict conservation isn't necessary. However, when we are on subcells we do maintain conservation both within the DG element and with any neighbors. Since the time stepper is used in regions where there are shocks, we need it to be conservative. However, we could give up conservation away from shocks. If there's a way to "switch" time steppers when either us or one neighbor is doing subcell so as to ensure conservation, that should work fine. Of  course, I have no idea how easy that is to do. Does that make sense?

Reply:
We should discuss this more sometime, but I've derailed this PR enough.

Back on topic, I think we've decided that there is no common mathematical term for the relation between the projection operators, at least in the h-refined case.  Mentioning that prolongation is interpolation is useful (although I'm not sure what the "exact" clarifier means), but maybe just leave out any description of the meaning of restriction beyond the already mentioned L_2 stuff.